hawkular-inventory-api/src/test/java/org/hawkular/inventory/api/test/QueryTest.java

```diff
@@ -113,4 +116,34 @@ public void testOptimizationRessurectsAfterNonCanonicalSection() throws Exceptio
         Assert.assertTrue(q.getFragments()[1].getFilter() instanceof Related);
         Assert.assertTrue(q.getFragments()[2].getFilter() instanceof With.CanonicalPaths);
     }
+
+    @Test
+    public void testQueryBuilder() {
+        CanonicalPath cp1 = CanonicalPath.fromString("/t;tenant1");
+        CanonicalPath cp2 = CanonicalPath.fromString("/t;tenant2");
+
+        // old query
+        Query qMetricsRelationships = Query.path().with(
+                With.path(cp1),
+                SwitchElementType.incomingRelationships(),
+                RelationWith.name("name")).get();
+        Query qTypesRelationships = Query.path().with(
+                With.path(cp2),
+                SwitchElementType.incomingRelationships(),
+                RelationWith.name("name")).get();
+        Query oldQuery = new Query.Builder().with(new PathFragment(new NoopFilter()))
+                .branch().with(qMetricsRelationships).done()
+                .branch().with(qTypesRelationships).done().build();
+
+        // new query
+        Query.Builder builder = new Query.Builder().path().with(new NoopFilter()).builder()
+                .branch().path().with(With.path(cp1), SwitchElementType.incomingRelationships(),
+                        RelationWith.name("name")).builder().done()
+                .branch().path().with(With.path(cp2), SwitchElementType.incomingRelationships(),
+                        RelationWith.name("name")).builder().done();
+
```

Comment:
Not sure I like the `.builder().branch()` and `.builder().done()` "bridges". Why not `.branch()` and `.done()` directly?